server/src/services/commits.ts

```diff
@@ -94,11 +94,23 @@ const addCommit = async (
     );
   }
 
-  return await commitStorage.addCommit({
+  return commitStorage.addCommit({
     ...DEFAULT_COMMIT_PAYLOAD,
     ...commitData,
     createdAt: new Date(),
   });
 };
 
-export default {getAllCommits, addCommit};
+/**
+ * Deletes the commit with the specified commitId.
+ * An error would be thrown if commit does not already exist, and if deletion fails.
+ * @param commitId Id of the commit to be deleted
+ */
+const deleteCommit = async (commitId: number) => {
+  const commit = await commitStorage.getCommit(commitId);
+  const commitListing = await listingStorage.getListing(commit.listingId);
+
+  return commitStorage.deleteCommit(commitId, commitListing.id);
```

Comment:
Why do you need to retrieve the listing object? You could have just done this?
```suggestion
  return commitStorage.deleteCommit(commitId, commit.listingId);
```

Reply:
@karenfrilya97 Yep you are right! Haha I forgot that I was halfway through implementing the services layer (I was gonna do some validation against listing but now I realised I don't have to) when I got distracted by implementing the datastore layer and it slipped my mind to come back to it! 🙈 🙈 Sorry for the confusion!

Reply:
@hzxcaryn Ohh hahahah no wonder! That happens to me, too 😄